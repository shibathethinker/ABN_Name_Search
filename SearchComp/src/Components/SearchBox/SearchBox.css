.customTextbox{
    width: 100%;
  border-top:none;
  border-left:none;
  border-bottom:solid 2px #2C3E50;
  font-family: 'Candara' ;
  height: 25px;
  font-size:20px;
}

.customTextbox:focus,.customTextbox:active{
border-bottom:solid 2px #4CAF50;
outline:none ;
}

.customTextbox+label{
  transition:all .2s ease-out;
}
.customTextbox+.lbl{
position: absolute;
top:0px;
left:0px;

font-size: 18px;
}

.customTextbox:focus+ .lbl, .customTextbox:valid+.lbl{
    top:-15px;  
    font-size:10px;
}

.customTextbox+.lbl_moved{
position: absolute;
left:0px;
top:-15px;
font-size: 10px;
}

.textBoxContainer{
position:relative;
height:50px;
top:20px;
width:50%;
}

.err{
  color:#E74C3C;
  font-size: 14px;
}

