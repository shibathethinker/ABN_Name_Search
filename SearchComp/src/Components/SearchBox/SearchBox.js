import React from 'react';
import    './SearchBox.css';


/**
 * Customizable Searchbox 
 * Uses React's PureComponent for performance considerations
 * Following parameters can be passed for configuration but are optional -
 * @param width
 * @param placeholder
 * @param  Rule  
 */


 export default class SearchBox extends React.PureComponent{
constructor(props)
{
super(props);
this.state={labelClass:'lbl'}
this.labelClicked=this.labelClicked.bind(this);
}

labelClicked(e)
{
this.setState({labelClass:'lbl_moved'});
}

render()
{
return(
    <React.Fragment>
<div className="textBoxContainer"  style={{width:this.props.width}}>
<input id='inputText'  data-testid='inputText' className="customTextbox" required  onChange={this.props.onTextChangedFunc} onBlur={this.props.checkIfValidInputFunc}></input>
<label className={this.state.labelClass} name='placeholder_label' onClick={this.labelClicked}>{this.props.placeholder}</label>
<label id='inputError' data-testid='inputError' className="err">{this.props.errorText}</label>
</div>
</React.Fragment>
)
}
}


