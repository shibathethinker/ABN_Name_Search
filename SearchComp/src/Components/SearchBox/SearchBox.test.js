import SearchBox from './SearchBox';
import global from '../../../test/setupTests'
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'


test('searchbox renders',()=>{
const wrapper=global.shallow(<SearchBox></SearchBox>)
expect(wrapper.length).toBe(1);
})

test('searchbox input!',()=>{
const dummyFunction = jest.fn();
const wrapper=global.shallow(<SearchBox  onTextChangedFunc={dummyFunction}></SearchBox>)
const textBox=wrapper.find('#inputText')
textBox.simulate('change',{target:{value:'p123469988997'}})
console.log(textBox.debug());
expect(dummyFunction).toHaveBeenCalledWith({target:{value:'p123469988997'}});
})

test('seachbox snapshot',()=>{
let shallow=new ShallowRenderer();
let instance=shallow.render(<SearchBox/>)
expect(instance).toMatchSnapshot()
})


