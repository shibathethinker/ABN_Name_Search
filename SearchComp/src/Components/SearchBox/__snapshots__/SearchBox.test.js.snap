// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seachbox snapshot 1`] = `
<React.Fragment>
  <div
    className="textBoxContainer"
    style={
      Object {
        "width": undefined,
      }
    }
  >
    <input
      className="customTextbox"
      data-testid="inputText"
      id="inputText"
      required={true}
    />
    <label
      className="lbl"
      name="placeholder_label"
      onClick={[Function]}
    />
    <label
      className="err"
      data-testid="inputError"
      id="inputError"
    />
  </div>
</React.Fragment>
`;
